cloud-init.tf

```diff
@@ -0,0 +1,28 @@
+data "template_cloudinit_config" "config" {
+  gzip          = true
+  base64_encode = true
+
+  part {
+    content_type = "text/cloud-config"
+    content      = <<-EOF
+      package_update: true
+      packages:
+        - docker.io
+        - docker-compose
+      groups:
+        - docker
+      users:
+        - default
+        - name: ubuntu
+          groups: docker
+      write_files:
+        - path: /etc/cwagentconfig/cwagentconfig.json
+          encoding: b64
+          content: ${base64encode(file(coalesce(var.cloudwatch_agent_config, "${path.module}/cwagentconfig.json")))}
+          owner: root:root
+          permissions: '0644'
+      runcmd:
+        - [sudo, docker, run, -d, --restart, always, -v, /etc/cwagentconfig:/etc/cwagentconfig, amazon/cloudwatch-agent:1.247347.6b250880]
```

Comment:
Just 2 things here:
  - why are we using `sudo` here? AFAICT cloud-init runs as root, I would have expected something with `runuser`; 😄 
  - wouldn't it be better to use `amazon/cloudwatch-agent:latest` as a image for the CloudWatch agent?

Reply:
- A sad copy-paste from before, will remove `sudo`
- I maybe saw once that `latest` broke some features in `cloudwatch-agent` image, even if I can't remember which version was. In general using `latest` can be dangerous if the image has breaking changes. Anyway I don't have strong opinions on using or non using latest, so if you suggest to use it, I can freely change it

Reply:
Let's be conservative and stick to a version then, the image does not seem to provide a decent minor version tag.